### DISCUSSION TITLE
options.json.lock

### DISCUSSION THREAD

**Discussion by BrooksUltra** [felddy/foundryvtt-docker#355] [Q&A]
I had an electrician out at my house who tripped a breaking cutting power to my Rasberry Pi where my Foundry was hosted. I have Foundry installed on an SSD using felddy's Docker image connected to an external website that my players can access.  When I attempted to get back into the website, I received an Error 522 indicating a problem with the host.  I did some digging and discovered the "options.json.lock" file was present in /Config.  I stopped the container and deleted the file, but upon restarting the container it created the "options.json.lock" again.  I deleted the container, the  /Config folder and repulled the image.  Once again upon starting the container it created "options.json.lock".  I grabbed the SSD, reformated it, reinstalled Docker and Portainer, pulled the felddy docker image and still am finding the "options.json.lock" in my /Config.  I'm at the end of my rope.

**Comment by felddy** [comment]
It is normal for the `options.json.lock` directory will exist once Foundry has been started.  

This is my `Config` directory before starting Foundry:

```console
❱ ls -l data/Config/
total 32
-rw-r--r--@ 1 markf  staff  128 May 26 16:43 admin.txt
-rw-r--r--@ 1 markf  staff  837 Aug 16  2021 license.json
-rw-r--r--@ 1 markf  staff  464 May 26 16:43 options.json
-rw-r--r--@ 1 markf  staff   23 Dec  8 16:44 storage.json
```

And this is what it looks like once Foundry starts:
```console
❱ ls -l data/Config/
total 32
-rw-r--r--@ 1 markf  staff  128 May 26 16:48 admin.txt
-rw-r--r--@ 1 markf  staff  837 Aug 16  2021 license.json
-rw-r--r--@ 1 markf  staff  464 May 26 16:48 options.json
drwxr-xr-x  2 markf  staff   64 May 26 16:48 options.json.lock/
-rw-r--r--@ 1 markf  staff   23 Dec  8 16:44 storage.json
```

I am not privy to the exact mechanism that Foundry uses to manage its locks, but the existence of that directory is normal anytime after Foundry has been started.  The `options.json.lock` directory will persist even when the node server is stopped. 

~Can you provide a link to the information that lead you down this path of investigation?~

I found [your thread](https://discord.com/channels/170995199584108546/486930822465716249/979481986672521226) on Discord: 

**🌈 Phoenix (he/him)**'s reply to you is correct:
> The lock file always exists while foundry is running and is there to prevent a 2nd instance writing to files a running foundry instance is using. Unfortunately containerized installations are not officially supported and you'll need to contact felddy for any support using their container

> Additionally 522 is a cloudflare specific error involving their proxy server which is also not officially supported.

I think that something else in your setup between Cloudflare and your Pi didn't recover correctly.   I would work through their documentation to see if helps: 
- [Troubleshooting Cloudflare 5XX errors](https://support.cloudflare.com/hc/en-us/articles/115003011431-Troubleshooting-Cloudflare-5XX-errors#522error)

**Reply by BrooksUltra** [reply]
I did a search on the Founrdy Discord server under #troubleshooting.  

https://discord.com/channels/170995199584108546/486930822465716249/972138400197009428

It is my understanding that if Foundry is not shut down properly, the options.json.lock file will prevent Foundry from opening.  The typical advise is to navigate to /Config and manually delete the file.  This seems to work for most people.  

However, if I delete the file while the container is active, it becomes unhealthy and throws an error indicating it is looking for the options.json.lock folder.  Likewise, if I stop the container, and delete the file, it appears again once the container is restarted.
